### DISCUSSION TITLE
SteamCMD Fatal Error: Steamcmd needs to be online to update.

### DISCUSSION THREAD

**Discussion by laurentksh** [mbround18/valheim-docker#527] [Help!] [answered]
Hello !
I'm having an issue where everytime I try to run the container it encounters an error with steamcmd:

```
valheim_1   | [Valheim][steam]: Password: (REDACTED)
valheim_1   | [Valheim][steam]: Running Install...
valheim_1   | [ODIN][INFO]  - Installing 896660 to /home/steam/valheim
valheim_1   | [ODIN][INFO]  - steamcmd found in path
valheim_1   | Redirecting stderr to '/home/steam/.steam/logs/stderr.txt'
valheim_1   | ILocalize::AddFile() failed to load file "public/steambootstrapper_english.txt".
valheim_1   | [  0%] Checking for available update...
valheim_1   | [----] Verifying installation...
valheim_1   | [  0%] Downloading Update...
valheim_1   | [  0%] Checking for available update...
valheim_1   | [----] !!! Fatal Error: Steamcmd needs to be online to update.     Please confirm your network connection and try again.
valheim_1   | threadtools.cpp (3294) : Assertion Failed: Illegal termination of worker thread 'Thread(0x0x56d4a320/0x0xf700ab'
valheim_1   | [ODIN][INFO]  - Exited with http code: 254
valheim_1   | cp: cannot stat '/home/steam/steamcmd/linux64/steamclient.so': No such file or directory
valheim_1   | [Valheim][steam]: Initializing Variables....
valheim_1   | [ODIN][ERROR] - Failed to find server executable! Please run `odin install`
```

I've tried to upgrade my distro to Debian 10 Buster as recommended by #481 , but that did not fix the issue.
I am not familiar with Linux/Debian so I might have done something wrong, but I tried to follow the installation process correctly.

Thanks !

**Comment by mbround18** [comment]
@laurentksh it looks like the container you were spinning up did not have network access. Can you confirm that your container has network access?

 !!! Fatal Error: Steamcmd needs to be online to update.

**Reply by laurentksh** [reply]
Hmm that's odd since I was able to pull the image and update to Debian 10 Buster, do you know how I could check for sure if i'm having connection issues ?

**Reply by mbround18** [reply]
You could try spinning up the container with DEBUG_MODE=1 

And swap the image to use the :dev version rather than latest. I have a pr up where it will console out all that stuff

**Reply by laurentksh** [reply]
Hey so I tried doing that but it did not solve the issue.
After that I tried to re-install Debian on my VPS and it finally worked !
It seems the issue was caused by the Debian image I used which was provided by my VPS' host. So not your project's fault at all :D
Thanks a lot for your help and sorry for the bother :)

**Comment by laurentksh** [comment, marked as answer]
To summarize, my issue was that the Debian installation I was using wasn't working properly (Docker containers could not reach the internet).

So I did the following:
1. Re-installed Debian using a proper image (Debian 11 Bullseye)
2. Installed Docker by following these two tutorials: [Install Docker on Debian](https://docs.docker.com/engine/install/debian/#install-using-the-repository) and [Install Docker compose](https://docs.docker.com/compose/install/).
3. Added my user to the docker group by executing this command: ` sudo usermod -aG docker $USER`
4. Created a valheim folder under /home/$USER
5. Created the docker-compose.yml file with the settings I wanted.
6. Ran `docker-compose up`
7. And the container could now reach the internet and download the Valheim server from Steam's CDN !

**Reply by mbround18** [reply]
Yay! :) Glad you got it up and running! :)
